### DISCUSSION TITLE
How to Route Any Docker Container Through VPN in Unraid

### DISCUSSION THREAD

**Discussion by rowdya22** [RobertDWhite/WhiteMatterTech#19] [Q&A] [answered]
I'm setting up unRAID for the first time and found your extremely helpful guide: 
https://whitematter.tech/posts/how-to-route-any-docker-container-through-vpn-in-unraid/

I was hoping to get some clarification. 

If my Port Mappings are "172.17.0.2:8080/TCP192.168.4.99:8085" then I need to remove them from my app container and [create them in the privoxyvpn](https://whitematter.tech/posts/how-to-route-any-docker-container-through-vpn-in-unraid/#adding-gui-ports-to-vpn-container) container. I tired that by selecting “Add another Path, Port, Variable, Label or Device” entering the name of my app container and having the container port be 8080 and the host port 8085. 

While the privoxyvpn container started as did my app container, I cannot access it at 192.168.4.99:8085. What am I doing wrong? I'd really like to route several containers through PIA for security but keep hitting this same problem.

**Comment by RobertDWhite** [comment]
Hi @rowdya22 . Would you mind posting some screenshots of the config and/or docker-config screen? I would be interested in seeing the setup.

**Reply by rowdya22** [reply, marked as answer]
Thanks for getting back to me! 
I actually found a solution to this over the weekend. For whatever reason, I cannot use the "Network Type" drop down to select my VPN container, I had to manually specify "--net=container:privoxyvpn" in the "Extra Parameters" field for each container I wanted to route through the VPN. 
I also had to specify VPN_INPUT_PORTS and VPN_OUTPUT_PORTS to match the various app ports in the privoxyvpn container. 
![image](https://github.com/RobertDWhite/WhiteMatterTech/assets/19266317/d714af49-f370-4c52-bd74-03d1287e8a47)
![image](https://github.com/RobertDWhite/WhiteMatterTech/assets/19266317/66da3d4d-40b1-461d-8b2b-b8edb764cac1)
![image](https://github.com/RobertDWhite/WhiteMatterTech/assets/19266317/3291c4d5-f19c-41a2-afd2-832e82ff2788)
